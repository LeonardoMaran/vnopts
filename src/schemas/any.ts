import { Schema, SchemaParameters } from '../schema';
import { ValidateResult } from '../types';

export class AnySchema extends Schema<any, SchemaParameters<any>> {
  public expected() {
    return 'anything';
  }
  public validate(): ValidateResult {
    return true;
  }
}
