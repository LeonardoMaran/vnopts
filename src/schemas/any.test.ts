import { eachHandler } from '../../tests/__helpers__/utils';
import { AnySchema } from './any';

const name = '<key>';
const value = '<value>';

describe.each`
  parameters             | input                | output               | hasWarnings
  ${{}}                  | ${{ [name]: value }} | ${{ [name]: value }} | ${false}
  ${{ validate: false }} | ${{ [name]: value }} | ${Error}             | ${false}
`('', eachHandler<AnySchema>(AnySchema, { name }));
