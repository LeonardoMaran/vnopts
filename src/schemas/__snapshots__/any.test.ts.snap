// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema {"name":"<key>","validate":false} with input {"<key>":"<value>"} 1`] = `"Invalid \\"<key>\\" value. Expected anything, but received \\"<value>\\"."`;
